### PR TITLE
I have created the API's and tested the objects using POSTMAN

### PR DESCRIPTION
**Pull request by Peace-Makinita**

### CODE REVIEW

File: crimeApi/index.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+//import express
+var express = require('express');
+
+//report Route to send images to the uploads folder
+var reportRoute = require('./route/report');
+
+//Image Route to send images to the uploads folder
+var imageRoute = require('./route/image')
+
```
Comment (pasher11) [inline crimeApi/index.js]: Please delete this duplicate line that's commented out

File: crimeApi/index.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+var fs = require("fs");
+
+//We need to connect to our database CrimeMap
+//1.  We need to include our mongodb module so we can us it.
+// 2. We create a server instance for MongoD the same way we did for node js
+// 3. Remember to create a database on MongoDB we called ours CrimeMapp
+//4. We can now connect to our database
+var mongo = require('mongodb'),
+Server = mongo.Server,
+// Db = mongo.Db;
```
Comment (pasher11) [inline crimeApi/index.js]: Is there a reason you left `// Db = mongo.Db;` in here?

File: crimeApi/index.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+})
+
+//We are creating an API route
+app.listen(port, function(){
+	console.log("App is running on " + port);
+})
+
+
+
+// C:\Users\PEACE\Desktop\openApp\data
```
Comment (pasher11) [inline crimeApi/index.js]: This line is only related to your local instance. Please make it more generalized.

File: crimeApi/index.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+var mongo = require('mongodb'),
+Server = mongo.Server,
+// Db = mongo.Db;
+
+
+//MongoDb has its own port number localhost://27017
+var server = new Server('localhost', 27017, {auto_reconnect: true});
+//var db = new Db('CrimeMap', server);
+
+//Show a message if we connected successfully to our mongodb server
```
Comment (pasher11) [inline crimeApi/index.js]: Why are there so many commented lines? Do you plan to include this at some point?

File: crimeApi/index.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+//We are creating an API route
+app.listen(port, function(){
+	console.log("App is running on " + port);
+})
+
+
+
+// C:\Users\PEACE\Desktop\openApp\data
+
+/*
```
Comment (pasher11) [inline crimeApi/index.js]: Good documentation. I suggest compressing it so there's not so many lines of space.

File: crimeApi/models/report.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+		required: "This information is required"
+	},
+
+	createDate: {
+		type: Date,
+		default: Date.now
+	}
+});
+
+//create a variable to store our schema
```
Comment (pasher11) [inline crimeApi/models/report.js]: What is happening w/ these commented lines?

File: crimeApi/route/image.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+//import express
+var express = require('express');
+
+//We need to import the route module
+var router = express.Router();
+
+/*We need to import Multer
```
Comment (pasher11) [inline crimeApi/route/image.js]: I think you can delete "We need to import Multer" and "please install the multer package". Lines 9 & 10 are what we actually need to know.

File: crimeApi/route/image.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+		cb(null, file.fieldname + '-' + Date.now() + '.jpg')
+	}
+});
+
+var upload = multer({storage: storage}).single('crimeImage');
+
+router.post('/', function (req, res) {
+	upload(req, res, function(err) {
+		if(err) {
+			//there was an error when uploading
```
Comment (pasher11) [inline crimeApi/route/image.js]: You're not going to report errors?!

File: crimeApi/route/report.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+//We need to import the route module
+var Router = express.Router();
+
+console.log('this is the report route');
+
+
+//This route is used to retrieve all the data
+Router.route('/')
+	.get( function(req, res){
+		// res.send({name:'me',age:50});
```
Comment (pasher11) [inline crimeApi/route/report.js]: Please clean this up.

File: crimeApi/route/report.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+		})
+		.catch(function(err){
+			res.send(err);
+		})
+	});
+
+
+//We create a new Route to send data
+//This is a post resquest
+Router.route('/createReport')
```
Comment (pasher11) [inline crimeApi/route/report.js]: It is obvious that this is a post request, no need for a comment

File: frontEndFiles/resources/js/reportForm.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
+// alert("Our js is working");
+function saveForm(form) {
+	var crimeInfo {
```
Comment (pasher11) [inline frontEndFiles/resources/js/reportForm.js]: Please format this consistently.

File: crimeApi/index.js [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+app.use(bodyParser.urlencoded({extended: true}));
+
+//Report Route so we can use it
+app.use('/api/reportRoute', reportRoute)
+app.get('/' ,  function(req, res){
+	res.send('Result sent from report')
+})
+
+//We need to use our Image Route
+app.use('/imageRoute', imageRoute)
```
Comment (Dibwe243) [inline crimeApi/index.js]: Did you manage to attend to this?

File: crimeApi/index.js (hunk 1 above)
Comment (Dibwe243) [reply crimeApi/index.js]: Was this change implemented?

**Pull request closed without merging**